components/brave_wallet/browser/zcash/rust/lib.rs

```diff
@@ -144,3 +276,119 @@ impl OrchardExtendedSpendingKey {
         address.to_raw_address_bytes()
     }
 }
+
+impl OrchardAuthorizedBundle {
+    fn raw_tx(self: &OrchardAuthorizedBundle) -> Vec<u8> {
+        self.0.raw_tx.clone()
+    }
+}
+
+fn create_orchard_builder_internal(
+    orchard_tree_bytes: &[u8],
+    outputs: Vec<OrchardOutput>,
+    random_source: OrchardRandomSource
+) -> Box<OrchardUnauthorizedBundleResult> {
+    use orchard::Anchor;
+    use zcash_primitives::merkle_tree::read_commitment_tree;
+
+    // To construct transaction orchard tree state of some block should be provided
+    // But in tests we can use empty anchor.
+    let anchor = if orchard_tree_bytes.len() > 0 {
+        match read_commitment_tree::<MerkleHashOrchard, _, { orchard::NOTE_COMMITMENT_TREE_DEPTH as u8 }>(
+                &orchard_tree_bytes[..]) {
+            Ok(tree) => Anchor::from(tree.root()),
+            Err(_e) => return Box::new(OrchardUnauthorizedBundleResult::from(Err(Error::from(OrchardBuildError::AnchorMismatch)))),
+        }
+    } else {
+        orchard::Anchor::empty_tree()
+    };
+
+    let mut builder = orchard::builder::Builder::new(
+        orchard::builder::BundleType::DEFAULT,
+        anchor);
+
+    for out in outputs {
+        let _ = match Option::from(orchard::Address::from_raw_address_bytes(&out.addr)) {
+            Some(addr) => {
+                builder.add_output(None, addr,
+                    orchard::value::NoteValue::from_raw(out.value), None)
+            },
+            None => return Box::new(OrchardUnauthorizedBundleResult::from(Err(Error::WrongOutputError)))
+        };
+    }
+
+    Box::new(OrchardUnauthorizedBundleResult::from(match random_source {
+        OrchardRandomSource::OsRng(mut rng) => {
+            builder.build(&mut rng)
+                .map_err(Error::from)
+                .and_then(|builder| {
+                    builder.map(|bundle| OrchardUnauthorizedBundleValue {
+                        unauthorized_bundle: bundle.0,
+                        rng: OrchardRandomSource::OsRng(rng) }).ok_or(Error::BuildError)
+                })
+        },
+        OrchardRandomSource::MockRng(mut rng) => {
+            builder.build(&mut rng)
+                .map_err(Error::from)
+                .and_then(|builder| {
+                    builder.map(|bundle| OrchardUnauthorizedBundleValue {
+                        unauthorized_bundle: bundle.0,
+                        rng: OrchardRandomSource::MockRng(rng) }).ok_or(Error::BuildError)
+                })
+        }
+    }))
+}
+
+fn create_orchard_bundle(
+    orchard_tree_bytes: &[u8],
+    outputs: Vec<OrchardOutput>
+) -> Box<OrchardUnauthorizedBundleResult> {
+    create_orchard_builder_internal(orchard_tree_bytes, outputs, OrchardRandomSource::OsRng(OsRng))
```

Comment:
What's the expected behavior here if the OS RNG isn't available? I just want to make sure it's going to error properly and we'll handle that error accordingly if that does occur.

Reply:
Ahh, found the answer to my own question. Errors are caught on line 327 and OsRng returns errors if they aren't supplied.